client/Lykke.Service.Assets.Client/ReadModels/InMemoryAssetsReadModelRepository.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Autofac;
+using Lykke.Service.Assets.Client.Models.v3;
+using Microsoft.Extensions.Caching.Memory;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lykke.Service.Assets.Client.ReadModels
+{
+    class InMemoryAssetsReadModelRepository : IAssetsReadModelRepository, IStartable
+    {
+        public const string AllKey = "AllAssets";
+
+        private readonly IAssetsService _assetsService;
+        private readonly IMemoryCache _cache;
+
+        public InMemoryAssetsReadModelRepository(IAssetsService assetsService, IMemoryCache cache)
+        {
+            _assetsService = assetsService;
+            _cache = cache;
+        }
+
+        public Asset TryGet(string id)
+        {
+            try
+            {
+                if (!_cache.TryGetValue(id, out Asset value))
+                    return null;
+                return value;
+            }
+            catch (System.InvalidCastException)
+            {
+                return null;
+            }
+        }
+
+        public IReadOnlyCollection<Asset> GetAll()
+        {
+            var ids = _cache.Get<List<string>>(AllKey);
+            return ids.Select(x => _cache.Get<Asset>(x)).ToArray();
```

Comment:
Is this operation thread safe?

Reply:
Well, it is written that MS-provided MemoryCache is entirely thread safe.